lib/puppet/catalog-diff/searchfacts.rb

```diff
@@ -43,9 +43,28 @@ def find_nodes_puppetdb(env)
                ['=', 'title', capit]]]]]],
         )
       end
-      json_query = URI.encode_www_form_component(query.to_json)
+      query
+    end
+
+    def find_nodes_puppetdb(env)
+      require 'puppet/util/puppetdb'
+      puppetdb_version = 'latest'
```

Comment:
Can we get the PuppetDB version from the REST API instead?

https://puppet.com/docs/puppetdb/6.0/api/meta/v1/version.html

Reply:
I just rebased my branch with the latest from master and added a web API call to get the PuppetDB version now so this should be solved now.